feat(@desktop/metrics): Add metrics management page

endulab opened the pull request:
Add new settings page for metrics management - Privacy and security
Add popup to enable/disable metrics on Onboarding and Privacy screens
Add MetricsStore in QML

Issue https://github.com/status-im/status-desktop/issues/15490

https://github.com/user-attachments/assets/8f60d102-0a11-4dbf-a7f2-319c0fb25ff5

https://github.com/user-attachments/assets/87e8c140-2bbe-43be-8ca7-25168d1d6679

endulab committed:
feat(@desktop/metrics): Add metrics management page

Add new settings page for metrics management - Privacy and security
Add popup to enable/disable metrics on Onboarding and Privacy screens
Add MetricsStore in QML

Issue #15490

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 35f1650e | [#1](https://ci.status.im/job/status-desktop/job/prs/job/macos/job/aarch64/job/package/job/PR-15584/1/) | 2024-07-15 13:46:25 | ~5 min | `macos/aarch64` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/macos/job/aarch64/job/package/job/PR-15584/1/consoleText) |

jrainville commented on ui/app/AppLayouts/Onboarding/views/KeysMainView.qml:
```diff
@@ -20,12 +23,13 @@ Item {
             button1.forceActiveFocus()
         }
-        enableCentricMetrics.checked = root.startupStore.isCentralizedMetricsEnabled()
+        metricsEnablePopup.visible = true
```
I think we only want to show this popup on the first run, but it's worth checking with @xAlisher

endulab replied:
It is shown only during creating new account.

jrainville commented on ui/imports/shared/popups/MetricsEnablePopup.qml:
```diff
@@ -0,0 +1,115 @@
+            }
+
+            AgreementSection {
+                title: qsTr("What we not receive:")
```
This sentence is not correct English :thinking: 
Is it supped to say `What we won't receive`?
Also, this one has a `:` but not the one above

endulab replied:
The text has changed.

jrainville reviewed (commented):
Looks good, just two questions

anastasiyaig commented:
@endulab @jrainville we need to change tests for this PR to merge, we will work on this ASAP but not sure if we can do that immediately, perhaps it could be done tomorrow but @Valentina1133 is already checking on that

Valentina1133 committed:
chore: fixes for new flow

endulab commented:
@jrainville @igor-sirotin @micieslak you can review again.

jrainville reviewed (approved):
Looks good. Nice work

Valentina1133 committed:
chore: fixes for new flow (#15632)

igor-sirotin reviewed (commented):
@endulab In your video the popup appears after `Get your keys` screen, but in Figma it's after `Welcome to Status`:
<img width="1175" alt="image" src="https://github.com/user-attachments/assets/1ae5a8d1-1009-4701-9546-853eff690448">

Were there some particular reasons for this? If not, I think it's better to have it after the first screen. And the code would be simpler.

Also, I guess if the user clicked `Not now`, we shouldn't show it once again if the user goes back onto previous page and then forward again. Not a very important, but might annoy users during onboarding.

endulab commented:
Because in my next task I refactored the code and also adjusted UI according to a new design, I will close this pr in favor of https://github.com/status-im/status-desktop/pull/15685/files
